### ISSUE TITLE
ReferenceError (Mottle is not defined)

### ISSUE THREAD

**Issue by jamesallardice** [sporritt/jsPlumb#437]
Using either the version published to npm or the (apparently more recent) download available on the website I get a ReferenceError when the page loads. This appears to be due to the fact that when [`Mottle.consume` is assigned](https://github.com/sporritt/jsPlumb/blob/1.7.10/lib/mottle-0.6.js#L537), `Mottle` is undefined.

`Mottle` only appears to be defined as a property on `this`, and not as a local variable, although I haven't yet looked into what the context will be at that point.

Am I missing something obvious?

**Comment by sporritt** [comment]
what script is throwing this error?

**Comment by jamesallardice** [comment]
Both `dist/js/jsPlumb-1.7.9-min.js` and `dist/js/jsPlumb-1.7.9.js`. The `package.json` file lists `dist/js/dom.jsPlumb-1.7.9.js` as the entry point but that file doesn't appear to exist.

**Comment by stones** [comment]
@jamesallardice I had the same issue when i was trying to use webpack and [this issue](https://github.com/sporritt/jsPlumb/issues/399) solved it for me.

**Comment by 4poc** [comment]
While this might be a workaround it doesn't really address the underlying issues with the npm package does it? At least the Mottle load error seem to stem from enabled strict mode:

```
(function (){
this.Mottle = function () {};
Mottle.foo = 'bar';
})(); // will work
(function (){
'use strict';
this.Mottle = function () {};
Mottle.foo = 'bar';
})(); // will fail: Uncaught TypeError: Cannot set property 'Mottle' of undefined
```

**Comment by sporritt** [comment]
```
(function (){
'use strict';
this.Mottle = function () {};
Mottle.foo = 'bar';
}).call(this);
```

this one works.

**Comment by 4poc** [comment]
@sporritt oh wow ok nevermind then

**Comment by sporritt** [comment]
i will fix this in mottle though. in the next version.

**Comment by sporritt** [comment]
fixed in Mottle 0.7.1, which is in jsplumb 2.1.0 dev.

**Comment by lllama** [comment]
Apologies for the bump but is there a planned release date for 2.1.0? Currently hitting this this issue using webpack.
